File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/banner/BannerUiState.kt
package com.hhp227.concafe.presentation.main.cafemanagement.banner

data class BannerUiState(
    val selectedTab: BannerTab = BannerTab.ACTIVE,
    val banners: List<BannerItem> = emptyList(),
    val pendingDeleteBannerId: String? = null
) {
    val filteredBanners: List<BannerItem>
        get() = banners.filter { it.tab == selectedTab }

    val pendingDeleteBanner: BannerItem?
        get() = banners.firstOrNull { it.id == pendingDeleteBannerId }

    companion object {
        fun empty() = BannerUiState()
    }
}

data class BannerItem(
    val id: String,
    val cafeId: String?,
    val title: String,
    val description: String,
    val periodDays: Int,
    val statusLabelKey: String,
    val tab: BannerTab,
    val accentColorHex: String,
    val imageIcon: String,
    val imageUrl: String? = null
)

enum class BannerTab(val labelKey: String) {
    ACTIVE("banner_section_active"),
    SCHEDULED("banner_section_scheduled"),
    ENDED("banner_section_ended")
}

File: iosApp/iosApp/Presentation/Main/CafeManagement/Banner/BannerUiState.swift
//
//  BannerUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/15.
//

import Foundation

struct BannerUiState {
    var selectedTab: BannerTab = .active
    var banners: [BannerItem] = []
    var pendingDeleteBannerId: String? = nil

    var filteredBanners: [BannerItem] {
        banners.filter { $0.tab == selectedTab }
    }

    var pendingDeleteBanner: BannerItem? {
        guard let pendingDeleteBannerId else { return nil }
        return banners.first { $0.id == pendingDeleteBannerId }
    }
}

struct BannerItem: Identifiable {
    let id: String
    let cafeId: String?
    let title: String
    let description: String
    let periodDays: Int32
    let statusLabelKey: String
    let tab: BannerTab
    let accentHex: String
    let imageIcon: String
    let imageUrl: String?
}

enum BannerTab: String, CaseIterable {
    case active = "banner_section_active"
    case scheduled = "banner_section_scheduled"
    case ended = "banner_section_ended"
}
